tests/phpunit/Logger/LoggerFactoryTest.php

```diff
@@ -0,0 +1,184 @@
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Tests\Logger;
+
+use Generator;
+use Infection\Configuration\Entry\Badge;
+use Infection\Configuration\Entry\Logs;
+use Infection\Console\LogVerbosity;
+use Infection\Logger\BadgeLogger;
+use Infection\Logger\DebugFileLogger;
+use Infection\Logger\LoggerFactory;
+use Infection\Logger\PerMutatorLogger;
+use Infection\Logger\SummaryFileLogger;
+use Infection\Logger\TextFileLogger;
+use Infection\Mutant\MetricsCalculator;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Filesystem\Filesystem;
+
+final class LoggerFactoryTest extends TestCase
+{
+    public function test_it_creates_no_logger_for_low_verbosity(): void
+    {
+        $factory = new LoggerFactory(
+            new MetricsCalculator(),
+            new Filesystem(),
+            LogVerbosity::NONE,
+            true,
+            true
+        );
+
+        $loggers = $factory->createLoggersFromLogEntries(
+            new Logs(
+                '/a/file',
+                '/a/file',
+                '/a/file',
+                '/a/file',
+                null
+            ),
+            $this->createMock(OutputInterface::class)
+        );
+
+        $this->assertCount(0, $loggers);
+    }
+
+    /**
+     * @dataProvider provideLogTypesAndClasses
+     */
+    public function test_it_creates_logger_for_log_type(Logs $logs, string $expectedLogClass): void
+    {
+        $factory = new LoggerFactory(
+            new MetricsCalculator(),
+            new Filesystem(),
+            LogVerbosity::NORMAL,
+            true,
+            true
+        );
+
+        $loggers = $factory->createLoggersFromLogEntries(
+            $logs,
+            $this->createMock(OutputInterface::class)
+        );
+        $this->assertCount(1, $loggers);
+        $this->assertInstanceOf($expectedLogClass, current($loggers));
+    }
+
+    public function provideLogTypesAndClasses(): Generator
+    {
+        yield [
+            new Logs('text', null, null, null, null),
+            TextFileLogger::class,
+        ];
+
+        yield [
+            new Logs(null, null, null, 'per_muator', null),
+            PerMutatorLogger::class,
+        ];
+
+        yield [
+            new Logs(null, 'summary_file', null, null, null),
+            SummaryFileLogger::class,
+        ];
+
+        yield [
+            new Logs(null, null, 'debug_file', null, null),
+            DebugFileLogger::class,
+        ];
+    }
+
+    public function test_it_creates_a_bade_logger(): void
```

Comment:
`test_it_creates_a_badge_logger` or `test_it_creates_a_badge_logger_on_normal_verbosity` since it appears to be important here

Reply:
Why is this one not included in `test_it_creates_logger_for_log_type` instead?